ordering/__init__.py

```diff
@@ -68,11 +68,13 @@ def __contains__(self, item: _T) -> bool:
         return item in self._labels
 
     def __iter__(self) -> Iterator[T]:
-        item = self._successors[self._start]
-
-        while not isinstance(item, _Sentinel):
+        successors = self._successors
+        item = successors[self._start]
+        end = self._end
+        while item is not end:
+            assert not isinstance(item, _Sentinel)
             yield item
-            item = self._successors[item]
+            item = successors[item]
```

Comment:
That's because it's copied verbatim from a local branch where I was originally trying to only use a single delimiter i.e. OrderedDict's `__root`, thus the double underscores just kinda stayed upon splitting it in two; in fact, initially, I did split it to `__start` and `__stop`. Is it just me or are `_start` and `_end` a bit of a miss-matched pair?